### PR TITLE
Formplayer url routing

### PR DESCRIPTION
**Pull request by benrudolph**
@dannyroberts @wpride here's the HQ side of being able to route to machines based on username and domain

### CODE REVIEW

File: corehq/apps/cloudcare/utils.py [hunk 1]
```diff
@@ -0,0 +1,8 @@
+from django.conf import settings
+
+
+def get_formplayer_url(domain, username):
+    return settings.FORMPLAYER_URL.format(
```
Comment (benrudolph) [inline corehq/apps/cloudcare/utils.py]: this url locally will just be `localhost:8080`, so the format will do nothing. when it's in prod the url will be `"/formplayer/{domain}/{username}"`

File: deployment/nginx/cchq_local_nginx.conf [hunk 2]
```diff
@@ -28,6 +34,14 @@ server {
         try_files $uri $uri/ @proxy_to_app;
     }
 
+    # uncomment for formplayer
+    # location ~ ^/formplayer/(?<domain>[\w-]+)/(?<username>[\w-\.@]+)/(?<route>[^\/]*) {
+    #     set $request_hash $domain$username;
+    #     proxy_pass http://formplayer/$route;
+    #     proxy_read_timeout 900s;
+    #     proxy_next_upstream_tries 1;
+    # }
```
Comment (benrudolph) [inline deployment/nginx/cchq_local_nginx.conf]: this is essentially what the nginx config would look like. the location capture pulls out the `domain`, `username` and the rest of the route (`route`). we take the domain and username and set it to a variable we'd like to hash. then we do a proxy pass with remainder of the route to get the correct result

**Comment by benrudolph** [comment]
my "too sneaky" alarm was going off a bit when writing this stuff. let me know if you think it's too clever

**Comment by benrudolph** [comment]
no longer needed

**Comment by snopoke** [comment]
If this is no longer needed has it been super seeded by something else? I
assume this was for some kind of load balancing?

**Comment by benrudolph** [comment]
this is all we ended up needing: https://github.com/dimagi/commcarehq-ansible/pull/513/files

**Comment by snopoke** [comment]
Nice!

**Comment by benrudolph** [comment]
😄

**Pull request closed without merging**